Marlin/src/MarlinCore.cpp

```diff
@@ -937,16 +937,18 @@ void minkill(const bool steppers_off/*=false*/) {
     // Reboot the board
     hal.reboot();
 
+  #elif defined(PLR_REBOOT_TIMEOUT)
+
+    for (uint16_t i = 0; i < PLR_REBOOT_TIMEOUT; i++) {
+      hal.watchdog_refresh();
+      delay(1000);
```

Comment:
I tried `delay` and `safe_delay` methods, but they seem freezing forever and system was restarted by watchdog after four seconds. The only working method I found was `delayMicroseconds`